Deprecate methods in PermissionService in preparation for dp-permissions-api integration

janderson2 opened the pull request:
This PR was originally by @janderson2 in 2022 - I (@lindenmckenzie) have brought it up to date with the latest changes - the original PR text is below:

### What

Remove and deprecate the PermissionsService.isPublisher and PermissionsService.isAdministrator methods respectively in preparation for the migration to using the dp-permissions-api.

In order to support this migration remove unnecessary usages of these methods and remove the `POST /service` endpoint used for generating service tokens via the API. This endpoint is not used in the environments anyway as access permissions block it. The documentation here and in the operations docs have been updated accordingly.

### How to review

Ensure that the changes are valid.

### Who can review

!me

rafahop commented on zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/Collections.java:
```diff
@@ -1020,7 +1020,7 @@ public boolean skipDatasetVersionsValidation(Long userKey, Long overrideKey, Ses
-        if (!permissionsService.isPublisher(session)) {
+        if (!permissionsService.canEdit(session)) {
             info().user(session)
                     .reason("publisher permissions required")
```
Would it make sense changing this log message?

rafahop commented on zebedee-cms/src/test/java/com/github/onsdigital/zebedee/permissions/service/JWTPermissionsServiceImplTest.java:
```diff
@@ -100,51 +99,59 @@ public void isPublisher_Session_NullGroup_ShouldReturnFalse() throws Exception {
 
         Session session = new Session(TEST_SESSION_ID, TEST_USER_EMAIL, sessionGroups);
-        assertFalse(jwtPermissionsService.isAdministrator(session));
+        assertFalse(jwtPermissionsService.isAdminUser(session));
```
This is now a protected method and probably not worth keeping these tests here?

It seems `isAdminUser` is only present in `JWTPermissionsServiceImpl` so to be used by its child class `PermissionsServiceImpl` (from `isAdministrator`) which will go at some point. Maybe we could move it to the child class and these tests to a test class for `PermissionsServiceImpl` testing the public `isAdministrator` instead?

And do something similar for the `isPublisher` too.

janderson2 committed:
Remove the /service endpoint

Remove the /service endpoint as it is the only remaining code using the
isAdministrator method of the PermissionsService that will not be
removed by the move to the JWT sessions and there is another more
commonly used method for generating service tokens anyway via command
line. The docs for how to do this have been updated.

janderson2 committed:
Remove the /service endpoint

Remove the /service endpoint as it is the only remaining code using the
isAdministrator method of the PermissionsService that will not be
removed by the move to the JWT sessions and there is another more
commonly used method for generating service tokens anyway via command
line. The docs for how to do this have been updated.

janderson2 committed:
Replace deprecated isPublisher with canEdit

Replace usages of deprecated PermissionsService.isPublisher method
with the PermissionsService.canEdit method in preparation of further
work on the PermissionsService.

janderson2 committed:
Replace deprecated isPublisher with canEdit

Replace usages of deprecated PermissionsService.isPublisher method
with the PermissionsService.canEdit method in preparation of further
work on the PermissionsService.

janderson2 committed:
Remove unnecessary, deprecated isAdministrator mock

janderson2 committed:
Remove unnecessary, deprecated isAdministrator mock

janderson2 committed:
Deprecate isAdministrator and remove isPublisher

Remove the isPublisher method from the PermissionsService interface
as it is only required internally now and this better prepares for the
upcoming work to migrate to using the permissions API.

Also deprecate the isAdministrator method as it will no longer be
required after the JWT session migration is complete.

janderson2 committed:
Deprecate isAdministrator and remove isPublisher

Remove the isPublisher method from the PermissionsService interface
as it is only required internally now and this better prepares for the
upcoming work to migrate to using the permissions API.

Also deprecate the isAdministrator method as it will no longer be
required after the JWT session migration is complete.

janderson2 committed:
Simplify code and improve error message

janderson2 committed:
Simplify code and improve error message

janderson2 committed:
Fix permissions service

Fix a few issues found during rebase:
* Pass the permissions API URL to the permissions service
* Remove unneccessary readlock
* Make isAdmin permission check more accurate by checking for the admin
  role rather than the users:update permission
* Use the actual user's groups for use in hasPermission check rather
  than assuming all users are admins

janderson2 committed:
Fix permissions service

Fix a few issues found during rebase:
* Pass the permissions API URL to the permissions service
* Remove unneccessary readlock
* Make isAdmin permission check more accurate by checking for the admin
  role rather than the users:update permission
* Use the actual user's groups for use in hasPermission check rather
  than assuming all users are admins

lindenmckenzie committed:
Remove deprecated isPublisher method

lindenmckenzie committed:
Remove deprecated isPublisher method

lindenmckenzie committed:
Remove duplicated configuration

lindenmckenzie committed:
Remove duplicated configuration

lindenmckenzie committed:
Correct comment typo

lindenmckenzie committed:
Correct comment typo

lindenmckenzie committed:
Remove service token guide

lindenmckenzie committed:
Remove service token guide

lindenmckenzie committed:
Re-add missing imports

lindenmckenzie committed:
Re-add missing imports

janderson2 reviewed (commented):
LGTM. @lindenmckenzie over to you to review

janderson2 reviewed (commented):
Still LGTM post rebase 👍
